apps/vs-agent/src/controllers/admin/verifiable/TrustService.ts

```diff
@@ -0,0 +1,415 @@
+      verificationMethodId,
+    )
+    return credential.jsonCredential
+  }
+
+  public async issueCredential(
+    type: 'jsonld' | 'anoncreds',
+    jsonSchemaCredential: string,
+    claims: JsonObject,
+    did?: string,
+  ) {
+    try {
+      // Check schema for credential
+      const { agent, didRecord } = await this.getDidRecord()
+      const jscData = await this.fetchJson<W3cCredential>(jsonSchemaCredential)
+      const subjectId = this.getCredentialSubjectId(jscData.credentialSubject)
+      const schemaData = await this.fetchJson<JsonObject>(mapToEcosystem(subjectId))
+      const parsedSchema =
+        typeof schemaData.schema === 'string' ? JSON.parse(schemaData.schema) : schemaData.schema
+      const subjectProps = parsedSchema?.properties?.credentialSubject?.properties ?? {}
+
+      const attrNames = Object.keys(subjectProps).map(String)
+      if (attrNames.length === 0) {
+        throw new HttpException(
+          `No properties found in credentialSubject of schema from ${jsonSchemaCredential}`,
+          HttpStatus.BAD_REQUEST,
+        )
+      }
+      validateSchema(parsedSchema, claims)
+
+      switch (type) {
+        case 'jsonld':
+          if (!did)
+            throw new HttpException(`Did must be present for json-ld creential`, HttpStatus.BAD_REQUEST)
+          const credential = await this.issueW3cJsonLd(agent, didRecord, did, jsonSchemaCredential, claims)
+          return { status: 200, didcommInvitationUrl: '', credential }
+        case 'anoncreds':
+          const credentialDefinitionId = await this.getCredentialDefinition(agent, attrNames)
+          const request = await agent.credentials.createOffer({
+            protocolVersion: 'v2',
+            credentialFormats: {
+              anoncreds: {
+                attributes: attrNames.map(name => {
+                  return { name, mimeType: '', value: JSON.stringify(claims[name]) }
+                }),
+                credentialDefinitionId,
+              },
+            },
+          })
+          const { url: longUrl } = await createInvitation({
+            agent,
+            messages: [request.message],
+          })
+
+          const shortUrlId = await this.urlShortenerService.createShortUrl({
+            longUrl,
+            relatedFlowId: request.credentialRecord.id,
+          })
+          const didcommInvitationUrl = `${this.publicApiBaseUrl}/s?id=${shortUrlId}`
+          return {
+            status: 200,
+            didcommInvitationUrl,
+            credential: { credentialExchangeId: request.credentialRecord.id },
+          }
+        default:
+          throw new HttpException(`Invalid credential type: ${type}`, HttpStatus.BAD_REQUEST)
+      }
+    } catch (error) {
+      this.handleError('issue', did ?? '', error, 'Failed to issue credential')
+    }
+  }
+
+  public async getCredentialDefinition(agent: VsAgent, attrNames: string[]) {
+    const schemaTag = generateDigestSRI(JSON.stringify(attrNames))
```

Comment:
Here, for Verifiable Trust use case it is important to link the AnonCreds object to the actual JSON Schema Credential, so the way to discover if the credential definition already exists should be the JSON Schema Credential URI rather than the claims used to create it.

This link is not purely internal to the issuer agent: it must also published as object metadata (you can do it in did:webvh method appending something like `relatedJsonSchemaCredential` to the attested resource metadata), because we need to use it in the VT resolver (e.g. verre) to be able to verify the permissions associated to that credential.